pkg/cluster/cluster.go

```diff
@@ -0,0 +1,293 @@
+package cluster
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/docker/docker/api/types/filters"
+	"github.com/submariner-io/armada/pkg/deploy"
+	"github.com/submariner-io/armada/pkg/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	dockertypes "github.com/docker/docker/api/types"
+	dockerclient "github.com/docker/docker/client"
+	"github.com/gobuffalo/packr/v2"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/submariner-io/armada/pkg/defaults"
+	apiextclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	kind "sigs.k8s.io/kind/pkg/cluster"
+	kinderrors "sigs.k8s.io/kind/pkg/errors"
+)
+
+// Create creates cluster with kind
+func Create(cl *Config, provider *kind.Provider, box *packr.Box, wg *sync.WaitGroup) error {
+	currentDir, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	configDir := filepath.Join(currentDir, defaults.KindConfigDir)
+	err = os.MkdirAll(configDir, os.ModePerm)
+	if err != nil {
+		return err
+	}
+
+	kindConfigFilePath, err := GenerateKindConfig(cl, configDir, box)
+	if err != nil {
+		return err
+	}
+
+	raw, err := ioutil.ReadFile(kindConfigFilePath)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Creating cluster %q, cni: %s, podcidr: %s, servicecidr: %s, workers: %v.", cl.Name, cl.Cni, cl.PodSubnet, cl.ServiceSubnet, cl.NumWorkers)
+
+	if err = provider.Create(
+		cl.Name,
+		kind.CreateWithRawConfig(raw),
+		kind.CreateWithNodeImage(cl.NodeImageName),
+		kind.CreateWithKubeconfigPath(cl.KubeConfigFilePath),
+		kind.CreateWithRetain(cl.Retain),
+		kind.CreateWithWaitForReady(cl.WaitForReady),
+		kind.CreateWithDisplayUsage(false),
+		kind.CreateWithDisplaySalutation(false),
+	); err != nil {
+		if errs := kinderrors.Errors(err); errs != nil {
+			for _, problem := range errs {
+				return problem
+			}
+			return errors.New("aborting due to invalid configuration")
+		}
+		return errors.Wrap(err, "failed to create cluster")
+	}
+	wg.Done()
+	return nil
+}
+
+// Destroy destroys a kind cluster
+func Destroy(clName string, provider *kind.Provider) error {
+	log.Infof("Deleting cluster %q ...\n", clName)
+	if err := provider.Delete(clName, ""); err != nil {
+		return errors.Wrapf(err, "failed to delete cluster %s", clName)
+	}
+
+	usr, err := user.Current()
```

Comment:
Here you are using the current user home dir but on create you were using current working dir. You need to consolidate to one.